include/NovelRT.h

```diff
@@ -0,0 +1,112 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_H
+#define NOVELRT_H
+
+#include <math.h>
+#include <type_traits>
+#include <map>
+#include <string>
+#include <glad.h>
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_mixer.h>
+#include <SDL2/SDL_surface.h>
+#include <SDL2/SDL_image.h>
+#include <SDL2/SDL_events.h>
+#include <string>
+#include <map>
+#include <typeinfo>
+#include <vector>
+#include <functional>
+#include <algorithm>
+#include <memory>
+#include <glm/glm.hpp>
+#include <glm/gtx/rotate_vector.hpp>
+#include <stdint.h>
+#include <iostream>
+#include <stdexcept>
+#include <fstream>
+#include <sstream>
+#include <limits>
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include "spdlog/spdlog.h"
+#include "spdlog/sinks/stdout_color_sinks.h"
+#include "spdlog/async.h"
+
+namespace NovelRT {
+  typedef void (*NovelUpdateSubscriber)(double deltaSeconds);
+  typedef class DebugService DebugService;
+  typedef class LoggingService LoggingService;
+  typedef class NovelRunner NovelRunner;
+}
+
+namespace NovelRT::Audio {
+  typedef std::map<std::string, Mix_Chunk*> SoundBank;
+  typedef std::map<std::string, Mix_Music*> MusicBank;
+  typedef std::map<std::string, int> ChannelMap;
+  typedef class AudioService AudioService;
+}
+
+namespace NovelRT::Graphics {
+  typedef std::conditional<sizeof(signed long) <= 4, GLuint, GLuint64>::type AdvanceInteger;
```

Comment:
What is `AdvanceInteger` for again?

Reply:
This is for the non-compliance of integer sizes and text character information. On different platforms, this will compile to the correct numeric type for that platform.

Reply:
I wonder if we should abstract that behind something....

If its just the `advance` (distance from this character to the next character in pixels) then 4 bytes is a reasonable size to expose (as we should never have a distance of > 4 billion pixels 😄).

We could also just expose it as `signed long`, but that puts the burden on the end user to normalize.

Reply:
(We probably also don't want to be tied to `GL*` types, for when we eventually support Vulkan and others).

Reply:
Well, the plan was I'd deal with that when we worked on a different render impl as I cannot think of a "nice" way to do it right now. Do you have a nice solution for this right now? If not I'll propagate it to then as I originally planned. 😄

Reply:
My suggestion is to basically make the `std::conditional` an internal implementation detail (as it is only needed for ogl anyways).

We would then decide to expose a consistent size to the end user. For example, we could choose:
* `int32_t`, should be big enough for all cases, will need extension on Unix 64-bit
* `int64_t`, will always be big enough, will need downcast on Windows and Unix 32-bit
* `intptr_t`, will always be big enough, will need downcast on Windows 64-bit
* `signed long`, exactly the right size, user will need to rationalize platform differences